Util/geometry.py

```diff
@@ -12,15 +12,46 @@ def __init__(self, p1, p2):
         self.p1 = p1
         self.p2 = p2
 
+    @property
+    def normalize(self):
+        return normalize(self.p2 - self.p1)
+
 
 class Area:
     def __init__(self, upper_left, lower_right):
-        self.a = upper_left  # -x, +y
-        self.b = lower_right # +x, -y
+        self.upper_left = upper_left   # -x, +y
+        self.lower_right = lower_right  # +x, -y
 
     def point_inside(self, p: Position) -> bool:
```

Comment:
Implementing `__contains__` would have been a cool API for an `Area` :  

```python
if Position(100,300) in Area(Position(0,0), Position(500,500)): ...
```

Reply:
Cool shit, I will do it!